sotodlib/hwp/hwp.py

```diff
@@ -665,20 +665,28 @@ def demod_tod(aman, signal=None, demod_mode=4,
                 lpf_cfg[k] = speed*float(v.split('*')[0])
 
     lpf = filters.get_lpf(lpf_cfg)
-        
+
+    # Create a RFFT object to reuse
+    n = fft_ops.find_superior_integer(aman.samps.count)
+    rfft = fft_ops.RFFTObj.for_shape(aman.dets.count, n, 'BOTH')
+
     phasor = np.exp(demod_mode * 1.j * aman.hwp_angle)
     demod = tod_ops.fourier_filter(aman, bpf, detrend=None,
-                                   signal_name=signal_name) * phasor
+                                   signal_name=signal_name, rfft=rfft).copy() * phasor
 
     # Filter the demodulated signal
     demod_aman = core.AxisManager(aman.dets, aman.samps)
     demod_aman.wrap("timestamps", aman.timestamps, axis_map=[(0, 'samps')])
     demod_aman.wrap("dsT", aman[signal_name], axis_map=[(0, 'dets'), (1, 'samps')])
-    demod_aman["dsT"] = tod_ops.fourier_filter(demod_aman, lpf, signal_name='dsT', detrend=None)
+    demod_aman["dsT"] = tod_ops.fourier_filter(demod_aman, lpf, signal_name='dsT', detrend=None, rfft=rfft).copy()
     demod_aman.wrap("demodQ", demod.real, axis_map=[(0, 'dets'), (1, 'samps')])
-    demod_aman["demodQ"] = tod_ops.fourier_filter(demod_aman, lpf, signal_name="demodQ", detrend=None) * 2.
+    demod_aman["demodQ"] = tod_ops.fourier_filter(demod_aman, lpf, signal_name="demodQ", detrend=None, rfft=rfft).copy() * 2.
```

Comment:
Once again, you don't need `.copy()`, because you're multiplying by 2.

However ... consider putting that factor of 2 into phasor, to begin with.  Then just use the same `[:]` trick as for dsT.